Assets/HoloToolkit/UX/Scripts/Buttons/Button.cs

```diff
@@ -0,0 +1,488 @@
+            /// </summary>
+            Interactive,
+            /// <summary>
+            /// Looking at button finger down
+            /// </summary> 
+            ObservationTargeted,
+            /// <summary>
+            /// Not looking at it and finger down
+            /// </summary>
+            Observation,
+            /// <summary>
+            /// Button in a disabled state
+            /// </summary>
+            Disabled,
+        }
+
+        /// <summary>
+        /// Current Button State
+        /// </summary>
+        [Header("Button")]
+        [Tooltip("Current State of the Button")]
+        public ButtonStateEnum ButtonState = ButtonStateEnum.Observation;
+
+        /// <summary>
+        /// Filter to apply for the correct button source
+        /// </summary>
+        [Tooltip("Filter for press info for click or press event")]
+        public InteractionSourcePressInfo ButtonPressFilter = InteractionSourcePressInfo.Select;
+
+        /// <summary>
+        /// If true the interactible will unselect when you look off of the object
+        /// </summary>
+        [Tooltip("If RequireGaze then looking away will unselect object")]
+        public bool RequireGaze = true;
+
+        /// <summary>
+        /// Event to receive button state change
+        /// </summary>
+        public event Action<ButtonStateEnum> StateChange;
+
+        /// <summary>
+        /// Event fired when tap interaction received.
+        /// </summary>
+        public event Action<GameObject> OnButtonPressed;
+
+        /// <summary>
+        /// Event fired when released interaction received.
+        /// </summary>
+        public event Action<GameObject> OnButtonReleased;
+
+        /// <summary>
+        /// Event fired when click interaction received.
+        /// </summary>
+        public event Action<GameObject> OnButtonClicked;
+
+        /// <summary>
+        /// Event fired when hold interaction initiated.
+        /// </summary>
+        public event Action<GameObject> OnButtonHeld;
+
+        /// <summary>
+        /// Event fired when hold interaction cancelled.
+        /// </summary>
+        public event Action<GameObject> OnButtonCancelled;
+
+        #endregion
+
+        #region Private and Protected Members
+        /// <summary>
+        /// Internal protected member for our default gizmo icon
+        /// </summary>
+        protected string _gizmoIconDefault = "HUX/hux_button_icon.png";
+
+        /// <summary>
+        /// Internal protected member for our gizmo selected icon
+        /// </summary>
+        protected string _gizmoIconSelected = "HUX/hux_button_icon_selected.png";
+
+        /// <summary>
+        /// Protected string for the current active gizmo icon
+        /// </summary>
+        protected string _gizmoIcon;
+
+        /// <summary>
+        /// Last state of hands being visible
+        /// </summary>
+        private bool m_bLastHandVisible = false;
```

Comment:
nit: inconsistent member prefixs? generally private members are `_camelCase` or `camelCase`

Reply:
I've been using m_camelCase for member vars with loose type prefixes.  I'll change it to remain consistent in MRTK.